examples/ensemble/plot_hgbt_regression.py

```diff
@@ -0,0 +1,397 @@
+    col_indices = np.random.choice(X.shape[1], num_missing_cells, replace=True)
+    X = df.drop(columns=["transfer", "class"])
+    X.iloc[row_indices, col_indices] = np.nan
+
+    hgbt.fit(X.iloc[train_0], y.iloc[train_0])
+    hgbt_predictions = hgbt.predict(X.iloc[test_0])
+    cv_results = cross_validate(
+        hgbt,
+        X,
+        y,
+        cv=ts_cv,
+        scoring="neg_root_mean_squared_error",
+    )
+    rmse = -cv_results["test_score"]
+    ax.plot(
+        hgbt_predictions[last_days],
+        label=(
+            f"missing_fraction={missing_fraction}, RMSE={rmse.mean():.2f} +/-"
+            f" {rmse.std():.2f}"
+        ),
+        alpha=0.5,
+    )
+ax.set(
+    title="Daily energy transfer predictions on data with MCAR values",
+    xticks=[(i + 0.25) * 48 for i in range(4)],
+    xticklabels=["Tue", "Wed", "Thu", "Fri"],
+    xlabel="Time of the week",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend()
+
+# %%
+# Support for quantile loss
+# =========================
+#
+# The quantile loss in regression enables a view of the variability or
+# uncertainty of the target variable. For instance, predicting the 5th and 95th
+# percentiles can provide a 90% prediction interval, i.e. the range within which
+# we expect the true value to fall with 90% probability.
+
+from sklearn.metrics import make_scorer, mean_pinball_loss
+
+quantiles = [0.95, 0.05]
+predictions = []
+X = df.drop(columns=["transfer", "class"])  # reset X
+
+fig, ax = plt.subplots(figsize=(12, 6))
+ax.plot(y.iloc[test_0].values[last_days], label="Actual transfer")
+
+for quantile in quantiles:
+    hgbt_quantile = HistGradientBoostingRegressor(
+        loss="quantile", quantile=quantile, **common_params
+    )
+    hgbt_quantile.fit(X.iloc[train_0], y.iloc[train_0])
+    hgbt_predictions = hgbt_quantile.predict(X.iloc[test_0])
+
+    predictions.append(hgbt_predictions)
+    cv_results = cross_validate(
+        hgbt_quantile,
+        X,
+        y,
+        cv=ts_cv,
+        scoring=make_scorer(mean_pinball_loss, alpha=quantile),
+    )
+    score = cv_results["test_score"]
+    ax.plot(
+        hgbt_predictions[last_days],
+        label=(
+            f"quantile={quantile}, pinball loss={score.mean():.3f} +/-"
+            f" {score.std():.3f}"
+        ),
+        alpha=0.5,
+    )
+
+ax.fill_between(
+    range(len(predictions[0][last_days])),
+    predictions[0][last_days],
+    predictions[1][last_days],
+    color=colors[0],
+    alpha=0.1,
+)
+ax.set(
+    title="Daily energy transfer predictions with quantile loss",
+    xticks=[(i + 0.25) * 48 for i in range(4)],
+    xticklabels=["Tue", "Wed", "Thu", "Fri"],
+    xlabel="Time of the week",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend()
+
+# %%
+# Keep in mind that those predicted percentiles are just estimations from a
+# model. One can still improve the quality of such estimations by:
```

Comment:
I think it worth to illustrate the problem of coverage on the under-estimation of the model a bit like in @ogrisel notebook: https://github.com/ogrisel/euroscipy-2022-time-series/blob/main/plot_time_series_feature_engineering.ipynb

Reply:
Addressed in [5cf52c2](https://github.com/scikit-learn/scikit-learn/pull/26991/commits/5cf52c27b3a328a0e65638036e9bd5db470472a7).